### PR TITLE
Create Yale Program on Financial Stability Resource Library translator

### PR DESCRIPTION
**Pull request by runkelcorey**
This translator scrapes the [YPFS repository](https://ypfs.som.yale.edu/) for documents, webpages, images, presentations, and video recordings related to financial crises.

### CODE REVIEW

**Comment by runkelcorey** [comment]
Selenium tests failed. Looks exactly like #2357 @optiprime faced. Need I recommit w/ `[skip ci]`?

`Translator server is running on http://localhost:8085
ElementNotInteractableError: element not interactable
  (Session info: chrome=89.0.4389.90)
    at Object.throwDecodedError (/home/runner/work/translators/translators/node_modules/selenium-webdriver/lib/error.js:550:15)
    at parseHttpResponse (/home/runner/work/translators/translators/node_modules/selenium-webdriver/lib/http.js:565:13)
    at Executor.execute (/home/runner/work/translators/translators/node_modules/selenium-webdriver/lib/http.js:491:26)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    at async thenableWebDriverProxy.execute (/home/runner/work/translators/translators/node_modules/selenium-webdriver/lib/webdriver.js:700:17)
    at async /home/runner/work/translators/translators/.ci/pull-request-check/selenium-test.js:118:3`

**Comment by adam3smith** [comment]
No need to recommit: we'll accept with failing CI if otherwise OK.

**Comment by runkelcorey** [comment]
So, could y'all merge this? I don't plan on making any other changes to it. Not sure if there's a lead time before adding new translators

File: ypfs.js [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+*/
+
+
+function detectWeb(doc, url) {
+	if (url.includes('/library/')) {
+		return 'document';
+
+	} else if (url.endsWith('.edu') || url.endsWith('.edu/') || url.includes('admin/content')) {
+		return 'multiple';
+	}
```
Comment (AbeJellinek) [inline ypfs.js:45]: Add a `return false` at the end of this function for consistency with other translators

File: ypfs.js [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+			ZU.processDocuments(articles, scrape);
+		});
+	} else {
+		scrape(doc, url);
+	}
+}
+
+
+function scrape(doc, url) {
+	var type = itemType(doc.querySelector('a[href*="ypfsresourcelibrary"]').href);
```
Comment (AbeJellinek) [inline ypfs.js:84]: Is there another selector that could work here? It feels somewhat likely that the first link on the page matching that string won't *always* be the one you want here.

File: ypfs.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+		Korean: 'kr',
+		Mongolian: 'mn',
+		Polish: 'pl',
+		Portugese: 'pt',
+		Slovene: 'sl',
+		Spanish: 'es',
+		Swedish: 'sv',
+		Thai: 'th',
+		Turkish: 'tr',
+	}[lang];
```
Comment (AbeJellinek) [inline ypfs.js:168]: Add `|| lang` here to handle languages not in this map, just in case.

File: ypfs.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
+{
+	"translatorID": "5e1f3e08-ca5f-4196-a662-6cf46b3cdaa7",
+	"label": "YPFS",
```
Comment (AbeJellinek) [inline ypfs.js:3]: Can you rename the file to `YPFS.js` (rather than `ypfs.js`) to match this label?

**Review by AbeJellinek — CHANGES_REQUESTED** [review]
Thanks, this looks good! Sorry for the delay in getting to it. We have a bit of a backlog that I'm working on getting through right now. I left a couple comments - could you address them before we merge?

File: ypfs.js (hunk 2 above)
Comment (runkelcorey) [reply ypfs.js]: At the risk of sounding arrogant, I am confident that a matching string will always be the one users want. `detectWeb` significantly limits the type of pages that `scrape` can be used on

**Review by runkelcorey — COMMENTED** [review]
Accepted three comments and rejected a fourth to change a selector. Thanks for the review!

**Review by AbeJellinek — APPROVED** [review]

**Comment by AbeJellinek** [comment]
OK, thanks. Merged your changes in. Thanks so much for your contribution!

**Pull request merged**